Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp

```diff
@@ -306,62 +306,77 @@ namespace AZ
+                        {
+                            m_brdfTexture = RPI::AttachmentImage::FindOrCreate(brdfImageAsset);
+                        }
+                    }
 
-                // Save a reference to the generated BRDF texture so it doesn't get deleted if all the passes refering to it get deleted and it's ref count goes to zero
-                if (!m_brdfTexture)
-                {
-                    const AZStd::shared_ptr<const RPI::PassTemplate> brdfTextureTemplate = RPI::PassSystemInterface::Get()->GetPassTemplate(Name("BRDFTextureTemplate"));
-                    Data::Asset<RPI::AttachmentImageAsset> brdfImageAsset = RPI::AssetUtils::LoadAssetById<RPI::AttachmentImageAsset>(
-                        brdfTextureTemplate->m_imageAttachments[0].m_assetRef.m_assetId, RPI::AssetUtils::TraceLevel::Error);
-                    if (brdfImageAsset.IsReady())
+                    if (!scene->GetRenderPipeline(AZ::Name(pipelineDesc.m_name)))
                     {
-                        m_brdfTexture = RPI::AttachmentImage::FindOrCreate(brdfImageAsset);
+                        RPI::RenderPipelinePtr brdfTexturePipeline = AZ::RPI::RenderPipeline::CreateRenderPipeline(pipelineDesc);
+                        scene->AddRenderPipeline(brdfTexturePipeline);
                     }
                 }
 
-                if (!scene->GetRenderPipeline(AZ::Name(pipelineDesc.m_name)))
+                // Load XR pipelines if applicable
+                if (xrSystem)
                 {
-                    RPI::RenderPipelinePtr brdfTexturePipeline = AZ::RPI::RenderPipeline::CreateRenderPipeline(pipelineDesc);
-                    scene->AddRenderPipeline(brdfTexturePipeline);
+                    for (AZ::u32 i = 0; i < xrSystem->GetNumViews(); i++)
+                    {
+                        const AZ::RPI::ViewType viewType = (i == 0)
+                            ? AZ::RPI::ViewType::XrLeft
+                            : AZ::RPI::ViewType::XrRight;
+                        const AZStd::string_view xrPipelineAssetName = (viewType == AZ::RPI::ViewType::XrLeft)
+                            ? "passes/XRLeftRenderPipeline.azasset"
+                            : "passes/XRRightRenderPipeline.azasset";
+
+                        if (!LoadPipeline(scene, viewportContext, xrPipelineAssetName, viewType, multisampleState))
+                        {
+                            return false;
+                        }
+                    }
                 }
 
+                // Apply MSAA state to all the render pipelines.
+                // It's important to do this after all the pipelines have
+                // been created so the same values are applied to all.
+                // As it cannot be applied MSAA values per pipeline,
+                // it's setting the MSAA state from the last pipeline loaded.
+                AZ::RPI::RPISystemInterface::Get()->SetApplicationMultisampleState(multisampleState);
```

Comment:
This is problematic. There is/was a bug where you couldnt set MSAA state after the pipeline is loaded and added to the scene. I had to revert this specific change due to that bug (https://github.com/o3de/o3de/pull/12359). Maybe it is fixed due to recent changes made by Vicky here https://github.com/o3de/o3de/pull/12700 but this needs to be confirmed by @VickyAtAZ and @dmcdiarmid-ly. If that bug is fixed I am ok with this change as that is how I intended it to work but if not we need to revisit this code.

Reply:
It's funny that I changed it to be the same code you had it in first place in #12359  :) 

Unfortunately without this change it lead to errors on Quest 2. Previously it was applying MSAA for default renderer only and not for XR pipelines. That causes errors when running on Quest2 because there is no default renderer and that means the MSAA state never gets applied and that leads to errors with the light culling pass mismatching the depth image descriptor having 1 sample set and the depth texture in the shader using TextureMS sampler (which expects image descriptors with samples >1 set).

Then I modified the code to set the MSAA at least once, which means on PC it'll be when loading the default renderer and on VR that'll be on the first XR pipeline (left eye pipeline, and not the right eye pipeline). That get rid of the error that it was rendering differently left eye and right eye, the shadows didn't appear in one of them.

I think I tested to always apply it and that didn't work either, but I can't remember what was the result.

The only way I got it to work for both PC and XR was by applying MSAA at the end after all pipelines were created.

The function `SetApplicationMultisampleState` internally does this

````
            // reinitialize pipelines for all scenes
            for (auto& scene : m_scenes)
            {
                for (auto& renderPipeline : scene->GetRenderPipelines())
                {
                    renderPipeline->GetRenderSettings().m_multisampleState = multisampleState;
                    renderPipeline->MarkPipelinePassChanges(PipelinePassChanges::MultisampleStateChanged);
                }
            }
````

Since it goes through all the render pipelines to set the MSAA state I thought it seems to make sense to apply it after the render pipelines have been created and added to the scenes.

@moudgils , @VickyAtAZ @dmcdiarmid-ly please advice on how to proceed. I'm happy to test other variants of when to set the MSAA state if you have other suggestions.

Thanks

Reply:
Lets first confirm if this bug was fixed. Can you check that DiffuseProbeGrid component is working properly within the Editor with these changes?. Try adding DiffuseProbeGrid volume on top of the shaderball in the default scene, the bottom of the shaderball should receive light from the ground. Ping @dmcdiarmid-ly if you need further clarification for testing this as he is the author of that component.

Reply:
Tested DiffuseProbeGrid component in a default level, I sent a screenshot to @dmcdiarmid-ly for verification and he confirmed: "The default irradiance color is white, so that looks like it's working.". I also tested changing the Sun's directional light color and it did affect the probes.

@moudgils please proceed to approve the PR if there is nothing else. Thanks.